common/primitives/src/messages.rs

```diff
@@ -0,0 +1,110 @@
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
+use sp_runtime::traits::AtLeast32BitUnsigned;
+use sp_std::{prelude::*, vec};
+use crate::msa::MessageSenderId;
+
+pub type SchemaId = u16;
+
+
+#[derive(Default, Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq)]
+pub struct Message<AccountId> {
+	pub data: Vec<u8>,		    	//  Serialized data in a user-defined schema format
+	pub signer: AccountId,	    	//  Signature of the signer
+	pub msa_id: MessageSenderId,    //  Message source account id (the original sender)
+	pub index: u16,		    		//  Stores index of message in block to keep total order
+}
+
+#[derive(Default, Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq)]
+pub struct MessageResponse<AccountId, BlockNumber> {
+	pub message: Message<AccountId>,
+	pub block_number: BlockNumber
+}
+
+#[derive(Default, Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq)]
+pub struct BlockPaginationRequest<BlockNumber> {
+	pub from_block: BlockNumber, 	// inclusive
+	pub from_index: u32,			// starts from 0
+	pub to_block: BlockNumber,   	// exclusive
+	pub page_size: u32,
+}
+
+impl<BlockNumber> BlockPaginationRequest<BlockNumber>
+	where BlockNumber: Copy + AtLeast32BitUnsigned
+{
+	pub const MAX_PAGE_SIZE: u32 = 10000;
+	pub const MAX_BLOCK_RANGE: u32 = 50000; // ~ 3 days
```

Comment:
I thought these numbers makes sense. Please let me know if you disagree.

Reply:
~7 days at 12 second block time right?

Reply:
My calculation was based on 6 sec per block but If the blocks are 12sec then it would be 7 days. I can update the comment